librepo/downloader.c

```diff
@@ -2344,11 +2345,15 @@ check_transfer_statuses(LrDownload *dd, GError **err)
                     target->tried_mirrors = g_slist_remove(target->tried_mirrors, target->mirror);
                     num_of_tried_mirrors = g_slist_length(target->tried_mirrors);
                 }
-
-                if (can_try_more_mirrors(dd, num_of_tried_mirrors))
+                gboolean complete_url_or_baseurl = complete_url_in_path || target->target->baseurl;
+                if (can_try_more(dd, num_of_tried_mirrors, complete_url_or_baseurl))
```

Comment:
It feels like this function does two independent things now and is no longer very helpful. Consider putting the contents of the function directly in place here? I don't insist on this one though.

Reply:
The function was renamed and added a description for the function. Hope that it helps.